pet-store-react-hooks/src/Home/Home.tsx

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { getAllPetsAsync, getPetKindsAsync } from '~infrastructure/api-client';
+import type { PetListItem } from '~infrastructure/api-types';
+import { ErrorMessage } from '~infrastructure/components/errorMessage/ErrorMessage';
+import { LoadingIndicator } from '~infrastructure/components/loadingIndicator/LoadingIndicator';
+
+import { PetsTable } from './PetsTable';
+
+import './Home.css';
+
+export function Home() {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | undefined>();
+  const [allPets, setAllPets] = useState<PetListItem[] | undefined>();
+  const [petKindsMap, setPetKindsMap] = useState<
+    Map<number, string> | undefined
+  >();
+
+  const hasFetchedPetKinds = useRef<boolean>(false);
+
+  const refreshPets = useCallback(async () => {
+    setError(undefined);
+    setLoading(true);
+
+    try {
+      const petsPromise = getAllPetsAsync();
+
+      if (!hasFetchedPetKinds.current) {
+        const petKinds = await getPetKindsAsync();
+
+        const map = new Map<number, string>();
+        for (const kind of petKinds) {
+          map.set(kind.value, kind.displayName);
+        }
+
+        setPetKindsMap(map);
+        hasFetchedPetKinds.current = true;
+      }
+
+      const pets = await petsPromise;
+      pets.sort((x, y) => (x.petId > y.petId ? -1 : 1));
```

Comment:
You know that you can do `pets.sort((x, y) => x.petId - y.petId);` right?

Reply:
Yeah, its not like i search everytime how to sort things in javascript :D